gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -155,7 +155,37 @@ message CreateProcessInstanceWithResultResponse {
   // JSON document
   // consists of visible variables in the root scope
   string variables = 5;
+}
 
+message EvaluateDecisionRequest {
+  // the unique key identifying the decision (e.g. returned from a decision
+  // in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the decision ID
+  string decisionId = 2;
+  // JSON document that will instantiate the variables for the decision; it
+  // must be a JSON object, as variables will be mapped in a key-value fashion.
+  // e.g. { "a": 1, "b": 2 } will create two variables, named "a" and
+  // "b" respectively, with their associated values. [{ "a": 1, "b": 2 }] would
+  // not be a valid argument, as the root of the JSON document is an array and
+  // not an object.
+  string variables = 3;
+}
+
+message EvaluateDecisionResponse {
+  // the unique key identifying the decision (e.g. returned from a decision
+  // in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the decision ID
+  string decisionId = 2;
+  // the version of the decision which was evaluated
+  int32 version = 3;
+  // JSON document that will instantiate the result of the decision; it
+  // will be a JSON object, as the result output will be mapped in a key-value
+  // fashion, e.g. { "a": 1 }.
+  string result = 4;
```

Comment:
❌ We should provide the entire decision evaluation (incl. audit trail) to the user.

We communicate the entire [decision evaluation record](https://github.com/camunda/zeebe/pull/11193/files#diff-fb50406855683626536b0276b2c5507ca075b526f5a50135541a0b83aa1ce123R66-R79) to the gateway. And we can [map the response with all those details immediately](https://github.com/camunda/zeebe/pull/11281/files#diff-e03e3d649c6deacb52e9a21698acfb42189984b640b97aac022097afb428f74aR156-R157) into the gateway's response. So, we should model the response schema in such a way that it can provide all the evaluation details. For the structure, we can just reuse the structure of the decision evaluation record.

The response should include: 
- the `decisionRequirementsKey` and `decisionRequirementsId` of the DRD that the evaluated decision is part of, 
- the name of the decision that was evaluated, 
- the `decisionOutput` (the final result of the evaluated decision), 
- and the `evaluatedDecisions` containing all the details about the evaluated decisions (audit), incl.
  - the decision type, 
  - evaluated inputs, 
  - matched rules, 
  - and decision output.

Reply:
@korthout makes sense. During the kick-off, we discussed adding the intermediate results in a separate ticket (https://github.com/camunda/zeebe/issues/11050). If I add all this information now, then this ticket would be redundant.

:thinking: Do you think that it makes sense to use different naming for the following properties?
* The final `decisionOutput` to be `result` (as it is)
* The (audit) evaluated decisions can maybe be placed in an `intermediateResults`, so users can immediately recognize why they are there.

Reply:
Ah I only remembered that we needed this, not that we decided to split it into several issues. Sorry about that.

>Do you think that it makes sense to use different naming

Is that something we discussed in the kickoff? I thought to remember that we decided on using the same structure of the existing records as the response.

Reply:
> Is that something we discussed in the kickoff?
We didn't really discuss the namings. We just discussed if "intermediate results" should be included or not. The decision was that we must include them.

The naming was never explicitly discussed except in examples in the ProductHub epic (see [this](https://github.com/camunda/product-hub/issues/69#issuecomment-1318782460)).